File: 1. Fundamental-Level/CSharp/04. HQC-December-2015/[HW]/03. [HW] Code-Documentation/Mass-Effect/Engine/Commands/Command.cs
namespace MassEffect.Engine.Commands
{
    using System;
    using MassEffect.Interfaces;
    using MassEffect.Exceptions;

    public abstract class Command
    {
        protected Command(IGameEngine gameEngine)
        {
            this.GameEngine = gameEngine;
        }

        protected IGameEngine GameEngine { get; set; }

        public abstract void Execute(string[] commandArgs);

        protected void IsAlive(IStarship starship)
        {
            if (starship.Health <= 0)
            {
                throw new ShipException(string.Format("{0}  is not alive.", starship.Name));
            }
        }
    }
}

File: 1. Fundamental-Level/Java/Advanced-OPP-July-2016/04. Enumerations&Annotations/src/problem10_11_12_13_InfernoInfinity/core/commands/Command.java
package problem10_11_12_13_InfernoInfinity.core.commands;

import problem10_11_12_13_InfernoInfinity.contracts.Engine;
import problem10_11_12_13_InfernoInfinity.contracts.Executable;

public abstract class Command implements Executable {

    private Engine engine;

    protected Command(Engine engine) {
        this.engine = engine;
    }

    public Engine getEngine() {
        return this.engine;
    }
}